### ISSUE TITLE
Multiple `allowedOwners` will not by default select the first one

### ISSUE THREAD

**Issue by knulpi** [backstage/backstage#12286]
## Expected Behavior

When having `allowedOwners` set in the configuration file of a template with more than one entry, the first owner should be selected, so that when I enter a repository name and click "Next", the form should advance to the next step.

## Actual Behavior

Although the select field shows the first configured org as selected, it seems not to be internally. When entering a repo name and clicking "Next", the form is displayed as invalid:

![image](https://user-images.githubusercontent.com/24552458/176179967-f13405a2-f17e-486c-a7f0-793930974f17.png)

After selecting another org and then re-selecting the first one, it will work.

## Steps to Reproduce

1. Set `allowedOwners` in a template to more than one entry
2. Open the template form in backstage and try to only enter a repo name
3. Click "Next"

## Context

Example template config excerpt:

```
allowedOwners:
  - org1
  - org2
```

## Suggested fix

I fixed this in our backstage instance by fixing [this line](https://github.com/backstage/backstage/blob/937f178db5737d2344bd5297d505161757012494/plugins/scaffolder/src/components/fields/RepoUrlPicker/RepoUrlPicker.tsx#L86) in the scaffolder to be
```typescript
    if (allowedOwners.length > 0) {
```

As it seems, this will set the default state to the first entry of the `allowedOwners`. Before, the snipped seems to only would have been executed for a lenght of 1.

## Your Environment

Tested with backstage release 1.1.0 and 1.3.0, same behavior. 

- Browser Information: tested on Firefox and Chromium

- Output of `yarn backstage-cli info`: 

```text
OS:   Linux 5.13.0-51-generic - linux/x64
node: v16.13.0
yarn: 1.22.17
cli:  0.17.2 (installed)

Dependencies:
  @backstage/app-defaults                          1.0.2
  @backstage/backend-common                        0.13.5
  @backstage/backend-tasks                         0.3.1
  @backstage/backend-test-utils                    0.1.24
  @backstage/catalog-client                        1.0.2
  @backstage/catalog-model                         1.0.2
  @backstage/cli-common                            0.1.9
  @backstage/cli                                   0.17.1
  @backstage/config-loader                         1.1.1
  @backstage/config                                1.0.1
  @backstage/core-app-api                          1.0.2
  @backstage/core-components                       0.9.4
  @backstage/core-plugin-api                       1.0.2
  @backstage/errors                                1.0.0
  @backstage/integration-react                     1.1.0
  @backstage/integration                           1.2.0
  @backstage/plugin-api-docs                       0.8.5
  @backstage/plugin-app-backend                    0.3.32
  @backstage/plugin-auth-node                      0.2.1
  @backstage/plugin-catalog-backend-module-github  0.1.3
  @backstage/plugin-catalog-backend                1.1.2
  @backstage/plugin-catalog-common                 1.0.2
  @backstage/plugin-catalog-graph                  0.2.17
  @backstage/plugin-catalog-import                 0.8.8
  @backstage/plugin-catalog-react                  1.1.0
  @backstage/plugin-catalog                        1.2.0
  @backstage/plugin-github-actions                 0.5.5
  @backstage/plugin-org                            0.5.5
  @backstage/plugin-permission-common              0.6.1
  @backstage/plugin-permission-node                0.6.1
  @backstage/plugin-permission-react               0.4.1
  @backstage/plugin-proxy-backend                  0.2.26
  @backstage/plugin-scaffolder-backend             1.2.0
  @backstage/plugin-scaffolder-common              1.1.0
  @backstage/plugin-scaffolder                     1.2.0
  @backstage/plugin-search-backend-module-pg       0.3.3
  @backstage/plugin-search-backend-node            0.6.1
  @backstage/plugin-search-backend                 0.5.2
  @backstage/plugin-search-common                  0.3.4
  @backstage/plugin-search-react                   0.2.0
  @backstage/plugin-search                         0.8.1
  @backstage/plugin-tech-radar                     0.5.12
  @backstage/plugin-techdocs-backend               1.1.1
  @backstage/plugin-techdocs-module-addons-contrib 1.0.0
  @backstage/plugin-techdocs-node                  1.1.1
  @backstage/plugin-techdocs-react                 1.0.0
  @backstage/plugin-techdocs                       1.1.1
  @backstage/plugin-user-settings                  0.4.4
  @backstage/release-manifests                     0.0.3
  @backstage/search-common                         0.3.4
  @backstage/test-utils                            1.1.0
  @backstage/theme                                 0.2.15
  @backstage/types                                 1.0.0
  @backstage/version-bridge                        1.0.1
```

**Comment by knulpi** [comment]
Yes, will do!

**Comment by knulpi** [comment]
fixed in #12471
